tests/acceptance/test_async.py

```diff
@@ -108,47 +108,51 @@ def example_task():
     assert len(jobs) == 1
 
 
-async def test_abort(async_app: app_module.App):
-    @async_app.task(queue="default", name="task1", pass_context=True)
-    async def task1(context):
-        while True:
-            await asyncio.sleep(0.02)
-            if await context.should_abort_async():
-                raise JobAborted
-
-    @async_app.task(queue="default", name="task2", pass_context=True)
-    def task2(context):
-        while True:
-            time.sleep(0.02)
-            if context.should_abort():
-                raise JobAborted
+@pytest.mark.parametrize("listen_notify", [False, True])
+@pytest.mark.parametrize("sync_task", [False, True])
+async def test_abort(async_app: app_module.App, listen_notify, sync_task):
+    if sync_task:
+
+        @async_app.task(queue="default", name="task1", pass_context=True)
+        def task1(context):
+            while True:
+                time.sleep(0.02)
+                if context.should_abort():
+                    raise JobAborted
+    else:
+
+        @async_app.task(queue="default", name="task1", pass_context=True)
+        async def task1(context):
+            while True:
+                await asyncio.sleep(0.02)
+                if context.should_abort():
+                    raise JobAborted
 
     job1_id = await task1.defer_async()
-    job2_id = await task2.defer_async()
+
+    polling_interval = 1
 
     worker_task = asyncio.create_task(
-        async_app.run_worker_async(queues=["default"], wait=False)
+        async_app.run_worker_async(
+            queues=["default"],
+            wait=False,
+            polling_interval=polling_interval,
+            listen_notify=listen_notify,
+        )
     )
 
     await asyncio.sleep(0.1)
     result = await async_app.job_manager.cancel_job_by_id_async(job1_id, abort=True)
     assert result is True
 
-    await asyncio.sleep(0.1)
-    result = await async_app.job_manager.cancel_job_by_id_async(job2_id, abort=True)
-    assert result is True
-
-    await worker_task
+    # when listening for notifications, job should cancel within ms
+    # if notifications are disabled, job will only cancel after polling_interval
+    await asyncio.wait_for(
```

Comment:
If I read this correctly, it means the test may take up to a second in normal conditions ?

I'd rather we keed tests shorter than this, we're already experiencing tests that are growing in duration, and if we can avoid adding tests that take a second when possible, that's better.